docs/federation/federation-directives-index.mdx

```diff
@@ -0,0 +1,31 @@
+| [`@edfs__natsSubscribe`](/federation/event-driven-federation/nats)           | Marks a subscription field that listens to one or more NATS subjects. Can include stream configuration.                                            |
+| `@external`                      | Indicates that a field (or all fields of a type) comes from another subgraph and isn’t resolved here.                                              |
+| `@extends`                       | Marks an object or interface as an extension of a type defined in another subgraph, when `extend` isn’t used.                                      |
+| `@inaccessible`                  | Hides a field or type from the API schema while keeping it in the supergraph for internal use.                                                     |
+| `@interfaceObject`               | Lets a subgraph contribute fields to an interface and all implementing types by defining an object abstraction.                                    |
+| `@key`                           | Declares one or more fields that uniquely identify an entity across subgraphs, enabling resolution.                                                |
+| `@link`                          | Connects an external spec to the schema and optionally imports types or directives for use in the subgraph.                                        |
+| [`@openfed__configureDescription`](/federation/directives/openfed__configuredescription) | Allows controlling how and whether descriptions are propagated to the federated schema. Supports overrides, hiding, and composition on extensions. |
+| [`@openfed__subscriptionFilter`](/federation/directives/openfed__subscriptionfilter)   | Applies a filter condition to a subscription field, restricting events based on field values or logical expressions.                               |
+| `@override`                      | Specifies that a field now resolves from the current subgraph instead of the one that previously defined it.                                       |
+| `@provides`                      | Indicates that specific fields can be resolved from this subgraph, but only at certain schema paths.                                               |
+| `@requires`                      | Declares that a field’s resolver depends on other fields from a different subgraph that must be fetched first.                                     |
+| [`@requiresScopes`](/federation/directives/requiresscopes)                | Restricts access to schema elements based on JWT scopes, enabling fine-grained authorization policies.                                             |
+| [`@shareable`](/federation/directives/shareable)                     | Marks a field or type as safe to define in multiple subgraphs. Allows shared resolution responsibilities.                                          |
```

Comment:
the meaning field could be resolved from multiple subgraphs, and it is determined by a planner from which subgraph to fetch them

If the choices are fully equal planner will pick first subgraph by order of appearence

If the shared field is a root query field and return type defines different fields both shared fields will be queried and merged together - but it is not recommended to use this approach because it will be responsibility of the subgraphs to keep data integrity

Overall when fields are shareable - it means that it should return the same value not matter from which subgraph

Reply:
Key fields are shared by definition/default